examples/applications/restapi/app.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import time
+from typing import List
+
+import bitcoinx
+
+from electrumsv.app_state import app_state
+from electrumsv.transaction import Transaction
+from electrumsv.wallet import UTXO, Abstract_Wallet
+from .handlers import ExtensionEndpoints
+
+
+class RESTAPIApplication:
+
+    def __init__(self):
+        self.logger = logging.getLogger("wallet-app")
+
+    def run_app(self):
+        self.logger.debug("entering application main loop")
+        try:
+            while app_state.async_.loop.is_running():
+                time.sleep(0.1)  # sole purpose is timely cleanup on KeyboardInterrupt
```

Comment:
I feel like this is also solved by waiting on some kind of event. I expect that the events these days allow for interruption by signals  like keyboard interrupt.  `threading.Event` will do the trick.
https://stackoverflow.com/questions/14421133/catch-keyboard-interrupt-in-program-that-is-waiting-on-an-event

Reply:
I had already been planning that as an improvement. Totally agree.

Reply:
eh... it looks like signal.pause() or one of the variants would do just fine for **unix** but for **windows**, this is still a problem that plagues many people afaict... surely someone, somewhere has solved it but there doesn't appear to be an easy way.

If I use threading.Event it just blocks the main thread (app.py runs on main thread) and never catches a KeyBoardInterrupt (because it's blocked at event.wait())

Reply:
Sure.